### ISSUE TITLE
Uniformed scoping for vlans, clusters and prefixes

### ISSUE THREAD

**Issue by voolean** [netbox-community/netbox#15396]
### NetBox version

v3.5.9

### Feature type

Change to existing functionality

### Proposed functionality

Some virtual objects like `prefixes` and `clusters` are bound to a site because they're using **SITE assignments** only.

Other objects like `vlan-group, ~~bgp, bgp-pool~~ are using **scope assignments**.

My proposed solution: Uniform virtual object scoping mechanism and apply scoping to vlans/cluster/prefix as well to support much more use-cases

### Use case

We have to migrate our `sites` to `child-regions` (or site-groups) but the virtual objects mentioned above are bound to a site only. When we want to migrate sites we have to move the prefix assignments as well. We can not assign these virtual objects to a region.

### Database changes

unsure

### External dependencies

none

**Comment by voolean** [comment]
- [#netbox slack thread](https://netdev-community.slack.com/archives/C01P0FRSXRV/p1709899390646309)
- [Allow prefixes to be assigned to a site group or region ](https://github.com/netbox-community/netbox/issues/6414)
- [Allow cluster to be assigned to a site group](https://github.com/netbox-community/netbox/issues/9969)
- [Assign virtualization Clusters to a Location](https://github.com/netbox-community/netbox/issues/7699) 
- [Support assigning prefixes/VLANs to locations](https://github.com/netbox-community/netbox/issues/6746)

**Comment by jeremystretch** [comment]
> Other objects like vlan-group,bgp, bgp-pool are using scope assignments.

There is no BGP or BGP pool model in NetBox; you're likely confusing these with models provided by a plugin.

> My proposed solution: Uniform virtual object scoping mechanism and apply scoping to vlans/cluster/prefix as well to support much more use-cases

This is going to need _much_ more detail to become an actionable feature request. Please invest some more effort draft a specific implementation instead of just linking to previous discussions.

**Comment by voolean** [comment]
Thanks Jeremy, I apologize for the confusion. I mistakenly associated BGP with another project.

I included links to previous discussions and issues mainly for reference, even though some of them may be outdated. Daniel Sheppard mentioned that this is something you're already looking into. I would be glad to provide additional details and elaborate.


Regarding the dynamic scoping for a prefix, I'm unsure about its functionality or specific implementation details. Instead of assigning a prefix to the site as...

`"site": { "id": 1, ..  },`

 ..I suggest making the following change:
```
"scope_type": "dcim.region",
"scope_id": 1,
"scope": {  "id": 1, ... },
```

**Comment by jeremystretch** [comment]
These concerns will be addressed under #6746 when/if that gets implemented, so I'm going to close this out as a duplicate.
